### ISSUE TITLE
Feature request: setting to enabled/disable sound and vibrate on disconnect

### ISSUE THREAD

**Issue by adamcmain** [JMRI/EngineDriver#753]
My throttle loses connection every 10~15 seconds but reconnects just fine nearly instantly.  It doesn't impact throttle control significantly, so I can live with it.  However, having the phone play a sound and vibrate nearly constantly is quite annoying.  

Please add a setting to disable the sound and vibration when the connection is lost or possibly a threshold where won't play until it's lost connection for more than a specific value.

This behavior was added with #511

**Comment by mstevetodd** [comment]
@adamcmain I've no issue with the request, but it's probably worthwhile to also investigate why your device is disconnecting regularly and fix it. You can contact me via email and I'll work with you on that.

@flash62au any thoughts on this request?

**Comment by flash62au** [comment]
@mstevetodd @adamcmain Yes, it is easy enough to add a preference to disable it, but I agree that I would be more concerned about why it is disconnecting so often.

**Comment by flash62au** [comment]
@adamcmain 
I have committed the change to add a preference to disable the sound and vibrate on disconnect. https://github.com/JMRI/EngineDriver/pull/758

It should be available on the release of the next beta.

**If you want to try it out now...**
I have created an apk here https://drive.google.com/file/d/16estVH21vJWAG6gEV_sXDNLHsI-yBNPv/view?usp=sharing
NOTE: I can't sign it with the real keys (Steve's keys) so if you want to try it you can download and install but you will need to uninstall the real version first (export your preference before you do, and import once this version is installed.)

**Comment by adamcmain** [comment]
Hi,  thanks for making the changes.  Any idea when the beta will include this option?

**Comment by mstevetodd** [comment]
> Hi, thanks for making the changes. Any idea when the beta will include this option?

"Soon" :)
I have "one more change", related to LnWi, I'm trying to get into this one. 
Unfortunately, DayJob and RealLife keep interfering!
It will be before the end of the year, with or without my "next" change.

**Comment by Alan-RMA49** [comment]
I'll jump in here as I've encountered a similar problem. As the club's social media guy and an avid instagramer in my (remaining) spare time, I use my phone a lot and to avoid using too much battery I have it set to lock automatically after 15 seconds. Whenever I'm using ED I increase the lock time to 60 seconds because I've found that whenever the phone locks, ED seems to lose the connection to JMRI and then takes a few seconds to reconnect (with the associated sound and triple vibration). Adam, could this be what you're encountering?

If that's the case, Android has a permission allowing an app to disable the lock screen under certain conditions (in ED's case it would probably need to be whenever it's in focus, but obviously with a warning that it'll use more battery). I think there's also a permission to maintain the network connection when the screen's locked, which might be a better solution. Having both probably be the ideal solution for end-users.

Of course I may be barking up the wrong tree here, but it's worth investigating.

Alan
